pennylane/debugging.py

```diff
@@ -105,3 +108,75 @@ def get_snapshots(*args, **kwargs):
         return dbg.snapshots
 
     return get_snapshots
+
+
+class PLDB(pdb.Pdb):
+    """Custom debugging class integrated with Pdb."""
+
+    __active_dev = []
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the debugger, and set custom prompt string."""
+        super().__init__(*args, **kwargs)
```

Comment:
# Note
We can pass a `skip` argument to avoid stepping into PennyLane internals. From the [docs](https://docs.python.org/3/library/pdb.html#pdb.Pdb),
>The _skip_ argument, if given, must be an iterable of glob-style module name patterns. The debugger will not step into frames that originate in a module that matches one of these patterns.

Reply:
Good call, I added this in the implementation of `breakpoint()` and left it general here.